A66-otel-stats.md

```diff
@@ -0,0 +1,542 @@
+
+A sample implementation of this approach is available in
+[gRPC Core](https://github.com/grpc/grpc/blob/v1.57.x/src/core/lib/channel/call_tracer.h).
+
+### Language-Specific Details
+
+Each language implementation will provide an API for registering an
+OpenTelemetry plugin. Overall, the APIs should have the following capabilities -
+
+*   Allow installing multiple OpenTelemetry plugins.
+*   Allow setting a
+    [MeterProvider](https://opentelemetry.io/docs/specs/otel/metrics/api/#meterprovider)
+    on individual plugins. Implementations should require a MeterProvider being
+    set. A MeterProvider not being set should result in a no-op. Some
+    OpenTelemetry language APIs have a global MeterProvider. gRPC
+    implementations should *NOT* fallback on this global.
+*   Optionally allow enabling/disabling metrics. This would allow optimizations
+    to avoid computation and collection of expensive stats within the gRPC
+    library. Note that even without this capability, users of OpenTelemetry
+    would be able to customize
+    [views](https://opentelemetry.io/docs/specs/otel/metrics/sdk/#view) through
+    the MeterProvider.
+*   Optionally allow setting of a OpenTelemetry plugin for a specific channel or
+    server, instead of setting it globally.
+*   Optionally allow setting of a map of constant attributes that are recorded
+    on all metrics associated with that plugin.
+
+Note that implementations of the gRPC OpenTelemetry plugin
+[should prefer](https://opentelemetry.io/docs/specs/otel/overview/) to only
+depend on the OpenTelemetry API and not the OpenTelemetry SDK.
+
+The [Meter](https://opentelemetry.io/docs/specs/otel/metrics/api/#get-a-meter)
+creation should use a `name` that identifies the library, for example,
+"grpc-c++", "grpc-java", "grpc-go". The `version` should be the same as the
+release version of the gRPC library, for example, "1.57.1".
+
+#### C++
+
+```c++
+class OpenTelemetryPluginBuilder {
+ public:
+  OpenTelemetryPluginBuilder();
+  // If `SetMeterProvider()` is not called, no metrics are collected.
+  OpenTelemetryPluginBuilder& SetMeterProvider(
+      std::shared_ptr<opentelemetry::metrics::MeterProvider> meter_provider);
+  // Methods to manipulate which instruments are enabled in the OTel Stats
+  // Plugin. The default set of instruments are -
+  // grpc.client.attempt.started
+  // grpc.client.attempt.duration
+  // grpc.client.attempt.sent_total_compressed_message_size
+  // grpc.client.attempt.rcvd_total_compressed_message_size
+  // grpc.server.call.started
+  // grpc.server.call.duration
+  // grpc.server.call.sent_total_compressed_message_size
+  // grpc.server.call.rcvd_total_compressed_message_size
+  OpenTelemetryPluginBuilder& EnableMetric(absl::string_view metric_name);
+  OpenTelemetryPluginBuilder& DisableMetric(absl::string_view metric_name);
+  OpenTelemetryPluginBuilder& DisableAllMetrics();
+  // If set, \a target_attribute_filter is called per channel to decide whether
```

Comment:
Nit: s/\a/a

Reply:
The \a was to just say that `target_attribute_filter` is a function argument